### PR TITLE
remove controlPlaneRef in KongTarget because it always refers to KongUpstream

### PR DESCRIPTION
**Pull request by randmonkey**
**What this PR does / why we need it**:
Because `KongTarget` always refers to a `KongUpstream`, so we do not need `controlPlaneRef` field so we remove it.
Also add `upstreamID` in `status.konnect` of `KongTarget`.
**Which issue this PR fixes**

Fixes #

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: api/konnect/v1alpha1/konnect_entity_status.go [hunk 1]
```diff
@@ -87,3 +87,13 @@ func (in *KonnectEntityStatusWithControlPlaneAndServiceRefs) SetControlPlaneID(i
 func (in *KonnectEntityStatusWithControlPlaneAndServiceRefs) GetControlPlaneID() string {
 	return in.ControlPlaneID
 }
+
+type KonnectEntityStatusWithControlPlaneAndUpstreamRefs struct {
+	KonnectEntityStatus `json:",inline"`
+
+	// ControlPlaneID is the Konnect ID of the ControlPlane this Route is associated with.
```
Comment (czeslavo) [inline api/konnect/v1alpha1/konnect_entity_status.go:94]:
```suggestion
	// ControlPlaneID is the Konnect ID of the ControlPlane this entity is associated with.
```

**Review by pmalek — COMMENTED** [review]
👍 

Looks good apart from what Greg  has pointed out.

File: api/konnect/v1alpha1/konnect_entity_status.go (hunk 1 above)
Comment (randmonkey) [reply api/konnect/v1alpha1/konnect_entity_status.go]: Copy pasta mistake :-(  fixed.

File: api/konnect/v1alpha1/konnect_entity_status.go [hunk 2]
```diff
@@ -87,3 +87,13 @@ func (in *KonnectEntityStatusWithControlPlaneAndServiceRefs) SetControlPlaneID(i
 func (in *KonnectEntityStatusWithControlPlaneAndServiceRefs) GetControlPlaneID() string {
 	return in.ControlPlaneID
 }
+
+type KonnectEntityStatusWithControlPlaneAndUpstreamRefs struct {
+	KonnectEntityStatus `json:",inline"`
+
+	// ControlPlaneID is the Konnect ID of the ControlPlane this entity is associated with.
+	ControlPlaneID string `json:"controlPlaneID,omitempty"`
```
Comment (pmalek) [inline api/konnect/v1alpha1/konnect_entity_status.go:95]: We can consider moving this field to a separate type and embedding it here via `inline` json struct tag. But that's a topic for another PR.

**Review by pmalek — APPROVED** [review]

File: api/konnect/v1alpha1/konnect_entity_status.go (hunk 2 above)
Comment (randmonkey) [reply api/konnect/v1alpha1/konnect_entity_status.go:95]: Does the issue https://github.com/Kong/kubernetes-configuration/issues/85 describe our requirement well?

**Pull request merged**